### ISSUE TITLE
Adding Cookie to download request

### ISSUE THREAD

**Issue by guillaumejay** [bezzad/Downloader#34]
Hi,

I'm trying to download files from volafile.org rooms, and to get files from it, when I use WebClient, I have to add
     client.Headers.Add(HttpRequestHeader.Cookie, "allow-download=1");
(without it, the request ends with  403 not Allowed error)
If i'm not mistaken, there no way to do that using RequestConfiguration.

**Comment by bezzad** [comment]
@guillaumejay Please test new version `2.1.0`. I added `CookieContainer` and `Headers` to `RequestConfiguration` class options.

**Comment by guillaumejay** [comment]
Thanks, it works !

**Comment by adreeeyan** [comment]
Hello @bezzad,

I think this change introduces a breaking change.
`Credentials` doesn't work anymore in the `DownloadConfiguration::RequestConfiguration`.
Instead, we now need to use `Headers`.

```c#
var headers = new WebHeaderCollection();
headers.Add(HttpRequestHeader.Authorization, "Basic " + Convert.ToBase64String(System.Text.Encoding.ASCII.GetBytes("username:password")))
new DownloadConfiguration()
{
    ParallelDownload = true, // download parts of file as parallel or not
    ...,
    RequestConfiguration = // config and customize request headers
    {
        Headers = headers
    }
};
```

**Comment by bezzad** [comment]
@adreeeyan Yes, this issue fixed now and you can test it from version v2.2.3 or later. Thanks for reporting this issue.

**Comment by adreeeyan** [comment]
@bezzad  Nice! Thank you for the prompt action.
